Lib/test/test_builtin.py

```diff
@@ -2138,6 +2138,22 @@ def test_bool_notimplemented(self):
         with self.assertRaisesRegex(TypeError, msg):
             not NotImplemented
 
+    def test_not_implemented(self):
```

Comment:
Should we add similar tests for Ellipsis while we're here? Maybe similar to `test_construct_singletons` just above here.